tracer/test/Datadog.Trace.Security.IntegrationTests/Rcm/AspNetCore5AsmToggle.cs

```diff
@@ -77,8 +77,16 @@ void CheckRequest(GetRcmRequest associatedRcmRequest, int fileNumberIfSecurityCa
                 if (EnableSecurity != null)
                 {
                     associatedRcmRequest.CachedTargetFiles.Should().BeEmpty();
-                    // Expected: APM_PRODUCT + the 3 ASM products
-                    associatedRcmRequest.Client.Products.Should().HaveCount(EnableSecurity == false ? 1 : 4);
+                    // Other products may be included, but none of the ASM ones should be
+                    var asmProducts = new[] { RcmProducts.Asm, RcmProducts.AsmData, RcmProducts.AsmDd };
+                    if (EnableSecurity == false)
+                    {
+                        associatedRcmRequest.Client.Products.Should().NotContain(asmProducts);
+                    }
+                    else
+                    {
+                        associatedRcmRequest.Client.Products.Should().Contain(asmProducts);
```

Comment:
maybe we could keep assessing it contains 3 products?

Reply:
The _problem_ is that it _doesn't_ contain 3 products 😅 It contains "whatever it should normally have" + 3 for ASM, so I think this achieves the same thing 🙂